### PR TITLE
Data contains dictionary if value is empty

### PR DESCRIPTION
**Pull request by iLLiCiTiT**
## Description
Use empty dictionary when `"data"` value is `None`.

## Additional information
When data are empty on entity server does return `None` instead of `"{}"`, added functionality makes `None` value as dictionary.

### CODE REVIEW

File: ayon_api/server_api.py [hunk 1]
```diff
@@ -8134,11 +8134,14 @@ def send_batch_operations(
...
-        entity_data = entity.get("data")
-        if (
-            entity_data is not None
-            and isinstance(entity_data, str)
-        ):
-            entity["data"] = json.loads(entity_data)
+
+        entity_data = entity["data"]
+        if entity_data is None:
+            entity_data = {}
```
Comment (BigRoy) [inline ayon_api/server_api.py:8142]: Isn't this quite similar as to just doing `entity_data = entity.get("data") or {}`

Comment (iLLiCiTiT) [reply ayon_api/server_api.py]: `"data"` should not be filled if are not available in entity.

Comment (BigRoy) [reply ayon_api/server_api.py]: I see - we may want to update the PR description then to explain that _with this PR_ it entity will also lack `"data"` completely whereas previously it may have been `None`. The PR isn't necessarily enforcing `data` to be `dict` but only enforcing that it will be - if it was existing in in the input data.

**Review by BigRoy — APPROVED** [review]
Didn't test but code looks good to me.

I did want to point out [the comment made here](https://github.com/ynput/ayon-python-api/pull/166#discussion_r1701797384) that the PR may do more than the PR description describes currently.

Also, not sure how to really test this? As in, when is data empty on the server? :)

**Comment by iLLiCiTiT** [comment]
> As in, when is data empty on the server? :)

All new entities that don't have anything in `"data"` :)
